Scan26 (#144)
pretix/pretix-docs: docs/guides/pretixscan/android.md

Reviewer: >     If you have finished setting up your event and then rented scan smartphones from us, then we have already installed and configured pretixSCAN and pretixPRINT for you on those devices.

War die temporale Implikation schon immer so rum?
Installiert sind die Apps auf 100% der Mietgeräte.
Für die Einichtung ist es nicht notwendig, **erst** eine fertige Veranstaltung zu haben und **dann** die Miete zu beauftragen. Es muss erst kurz vor Versand fertig sein. 

>     If you are renting scan smartphones from us, then we have already installed pretixSCAN and pretixPRINT for you on those devices. All relevant apps are configured for you unless communicated otherwise.

oder so; ich weiß schon, dass Vale da meckern wird.
Oder eben eine Formulierung mit "normally" wie unten bei PRINT.

# pretixSCAN (Android)

The Android version of [pretixSCAN](https://pretix.eu/about/en/scan) is an application that helps you handle the check-in at your events.
The primary function of pretixSCAN is to validate tickets, but it also offers many other features, such as:

 - searching for participants manually if they do not have their ticket on hand
 - printing attendee badges
 - quickly viewing attendance statistics

This article tells you how to install pretixSCAN for **Android** and how to make use of its features.

## Prerequisites

pretixSCAN is intended for use with an event hosted with pretix.

You need to have access to a device running Android version 7 or newer.
See our [Android version support policy](https://docs.pretix.eu/en/latest/user/android-version-support.html#pretixscan) for more information.

> You also need access to the pretix backend in order to set up pretixSCAN on your device.

Reviewer 2: ```suggestion
You also need access to the pretix backend or have been supplied with a qr code/access token generated by someone having access to the pretix backend.
```
Sometimes organizers also hand out the access tokens to other ppl for the installation of pretixSCAN, especially in the context of  amateur/volunteer organizations if ppl use their own smartphone to scan tickets. In this case there's no access to the pretix backend needed.

Thus I would write sth like this:
You also need access to the pretix backend or have been supplied with a qr code/access token generated by someone having access to the pretix backend.

## General usage

This section guides you through the basic usage of pretixSCAN on Android.
This includes the following steps:

 - [installing the app](#installation-and-setup) and setting it up
 - [connecting to the backend](#connecting-the-device-to-the-pretix-backend)
 - performing basic [entry scanning](#entry-scanning)
 - searching for [attendee data](#searching-for-attendee-data)
 - enabling [badge printing](#enabling-badge-printing)

> This section will guide you through those steps in detail.

Reviewer 2: ```suggestion
```
Since this is a repeat of basically the same sentence as above I would delete that sentence.

You can install pretixSCAN on your Android device [through the Google Play Store](https://play.google.com/store/apps/details?id=eu.pretix.pretixscan.droid) like any other app.
If your device does not have access to the Google Play Store, you can download the latest version from our [pretix Marketplace](https://marketplace.pretix.eu/products/pretixscan-android/versions).
On devices manufactured by Sunmi, you can also find the app in the Sunmi App Store.

![Screen titled "Welcome to pretixSCAN", asking you to connect the device to the backend.](../../assets/screens/scan/welcome.png "Welcome to pretixSCAN screenshot")

When you launch pretixSCAN for the first time, confirm that you understand the privacy and security implications of storing attendee data on your device.
You have to grant pretixSCAN access to the device's camera in order to use the core functionality of validating tickets.

!!! Note
    Our scan smartphones have both a camera and a scanner.
    The scanner works much more quickly and reliably than the camera.
    Thus, you should always use the scanner at your check-in.

>     If you are using a device without a scanner, then you should use the camera instead.

Reviewer 2: >Our scan smartphones have both a camera and a scanner.
    The scanner works much more quickly and reliably than the camera.
    Thus, you should always use the scanner at your check-in.

Because it happens more often than one might think, that customers use the camera instead of the much better scanner: Would it be an option to either include an annotated image of a scan smartphone showing the position of the scanner and the camera? Or describe it in more detail ('scanner points forward, dedicated scan buttons on the left and right hand side of the device')?

    In order to confirm that the device uses the right setting, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner, and then tap :btn:Settings:.
    Scroll to the section titled "User interface".
    If your device has a scanner, uncheck the box next to "Use device camera".
    If your device does have a scanner, check that box.

Once you have granted permission, pretixSCAN will start accessing the scanner or camera instantly.
It will also display a text box telling you to create a new device in our organizer account in the pretix backend.
The next section is going to tell you how to do that.

### Connecting the device to the pretix backend

!["Connect a new device" page, showing options for the name of the device, the events it should have access to, its security profile, and gate.](../../assets/screens/scan/connect-new-device.png "Connect new device screenshot")

Open the [pretix backend](https://pretix.eu/control/) and navigate to :navpath:Your organizer → :fa3-mobile-phone: Devices:.
Click the :btn-icon:fa3-plus: Connect a device: button.
Give the device a unique and recognizable name such as "Entrance B phone 1".

> You can authorize the device for all events or limit its access to certain events only.

Reviewer 2: > You can authorize the device for all events or limit its access to certain events only.

Sometimes customers forget that they just gave access to a certain event in the backend and then wonder after some time why the 'scanners don't work anymore'. Thus I would somehow advice users: "Bear in mind that you have to update this list in the backend if you don't grant the devices access for all events and you want to scan tickets for new events with these event restricted devices!"

Open the "Security profile" drop-down menu and select `pretixSCAN`.

!["Connect to device" page, displaying instructions for installing pretixSCAN, a QR code, and a token.](../../assets/screens/scan/connect-device.png "Connect to device screenshot")

Once you click the :btn:Save: button, pretix redirects you to a page displaying a QR code.

> Open pretixSCAN and scan the QR code with the camera or hardware scanner.

Reviewer 2: > Open pretixSCAN and scan the QR code with the camera or hardware scanner.

Maybe add that it's also possible and viable to print/save the qr code/token and hand it out to other ppl? This way one can prepare a bunch of devices for amateur/volunteer organizations if ppl want to use their own smartphones?

Then, tap :btn:Manual setup:.
Enter the system URL and token displayed in the pretix backend below the QR code.

![Screen in pretixSCAN, asking you to choose the event for which you want to scan tickets.](../../assets/screens/scan/select-event.png "Select event screenshot")

Once you scanned the QR code or entered the token successfully, the app will ask you to select the event for which you want to scan tickets.
Your next step depends on the type of event for which you want to scan tickets.

 - **Single event:** Select the event in the list or in the calendar and tap the :btn:OK: button.
   Then, select a check-in list and tap the :btn:OK: button again.

 - **Event series:** Select any of the dates from that event series.
   Tap the :btn:OK: button, select the check-in list, and tap the :btn:OK: button again.

 - **Multiple events or series:** Enable expert mode.
   See [Scanning tickets for multiple events](#scanning-tickets-for-multiple-events).

The app will now land you on the main screen and start downloading event data from the server in the background.

### Entry scanning

> The app should now be in entry scanning mode.

Reviewer 2: ```suggestion
By default the app should now be in entry scanning mode.
```

If it is, then it displays a pictogram of a gray box with an arrow pointing to the right :fa3-sign-in: in the top right corner.
If the pictogram is yellow and the arrow points to the left :fa3-sign-out:, that means that the app is in "exit scanning" mode.
Switch to entry scanning by tapping the three-dot button :btn-icon:fa3-ellipsis-v:: and :btn:Switch to entry scanning: in the popup menu.

pretixSCAN will scan any QR code or barcode it can detect via the device's camera or scanner.

> Point the device at the code you want to scan.

Reviewer 2: ```suggestion
If your're using a device with a scanner one has to point the upper edge of the device at the code you want to scan and push a button on the left/right side of the device. If you're using the camera of the device just point it at the code you want to scan.
``

The app will then check the code against the selected check-in list on the server.
There are three possible results:

 1. If the ticket is valid and allows entry under the current conditions, pretixSCAN displays a green box titled "Valid ticket".

>  Underneath the green box, it lists the ticket type, order number, and position number.

Reviewer 2: ```suggestion
 1. If the ticket is valid and the specified conditions of the check in list are met, pretixSCAN displays a green box titled "Valid ticket".
 Underneath the green box, it lists the ticket type, order number, and position number.
```

Author: Counterproposal: 
```suggestion
 1. If the ticket is valid and the check-in rules allow entry under the current conditions, pretixSCAN displays a green box titled "Valid ticket".
```
More precise, plus it avoids the passive voice

Reviewer 2: I fully agree. Sounds much better.

 2. If the ticket is valid but has already been checked in and not checked out again, pretixSCAN displays a yellow box titled "Ticket already used".
 Underneath the yellow box, it lists the ticket type, order number, and position number, and also the date and time when the ticket was first scanned.
 3. In all other cases, pretixSCAN displays a red box with a title explaining the problem.
 Explanations can include, among others, "Invalid ticket", "Order canceled", or "Entry not allowed".

![Scanning a ticket in pretixSCAN. A green box appears at the top of the screen, displaying a checkmark, the text: "Valid ticket (Speaker ticket)" and the order code suffixed with "-1".](../../assets/screens/scan/valid.png "Valid ticket screenshot")

If none of these things happen, then the code has not been scanned.
Try adjusting the angle and distance between scanner and ticket or improving ambient lighting.
You can turn the device's flashlight and automatic focus on or off with the buttons at the bottom of the screen.

> If you have trouble scanning a code from a phone screen, ask the customer to increase screen brightness to maximum.

Reviewer 2: ```suggestion
If you have trouble scanning a code from a phone screen, ask the customer to increase screen brightness to maximum and/or tilt the scanning device by 90 degrees left or right in case the light of the display is polarized.
```

### Searching for attendee data

An alternative to scanning a ticket code is searching for an attendee in pretixSCAN.
If an attendee arrives at the check-in without a ticket, but claiming that they have bought one, you can search for their data.
Tap the :btn-icon:fa3-search:: button at the top of the screen.
In the search field, enter the attendee's name, order code, ticket secret, or email address.
Tap the matching result in the list.

The app will now check the validity of the ticket and return the same results as if it was scanning the ticket.

### Enabling badge printing

Badge printing only works if you have enabled the "Badges" plugin for the current event and the ticket you are scanning has a badge layout selected.
In order to enable badge printing in pretixSCAN, you have to first install our additional app pretixPRINT on the same device and configure a connection to a printer.

!!! Note
    pretixPRINT comes preinstalled and configured on the scan smartphones that are available for rent on our website.

>     Thus, you will not have to install and configure pretixPRINT on a scan smartphone that you rented from us.

Reviewer 2: ```suggestion
    Thus, normally you will not have to install and configure pretixPRINT on a scan smartphone that you rented from us.
```
I would add 'normally' since there are some edge cases (change of ip addresses due to dhcp lease time or whatever) that could one force to call up pretixPRINT and change the ip address...

Scroll down to the section titled "Badges" and check the box next to "Print every badge twice".

### Switching to a different event or check-in list

If you want to switch to a different event or to a different check-in list for entry scanning, tap the name of your event at the top of the screen.
Select the event from the calendar or list and tap the :btn:OK: button.
Select the check-in list and tap the :btn:OK: button again.

If you want to scan tickets for multiple events at the same time, you should enable **expert mode**.
The same applies if you want to scan tickets for multiple dates in an event series which all have their own check-in lists.
See [Scanning tickets for multiple events](#scanning-tickets-for-multiple-events) for more information.

## Advanced usage

This section covers the following advanced use cases of pretixSCAN on Android:

 - [exit scanning](#exit-scanning)
 - [locking the settings](#locking-the-settings)
 - Letting attendees [scan their own tickets](#letting-attendees-scan-their-own-tickets) using kiosk mode
 - [offline scanning](#offline-scanning)
 - scanning tickets for [multiple events](#scanning-tickets-for-multiple-events)

### Exit scanning

> If you want to keep track of ticket holders leaving your event, you can use pretixSCAN's exit mode.

Reviewer 2: ```suggestion
There are two scenarios to use exit mode: Either if you want to keep track of ticket holders leaving your event, you can use pretixSCAN's exit mode or if you want to sell tickets for the same event again, by scanning tickets and freeing up the quota (you have to enable that option in the corresponding quota of the product).
``

> This is useful if you want to allow ticket holders to enter your event multiple times, but only if the same ticket is not currently checked in to the event.

Reviewer 2: ```suggestion
The first scenario is useful if you want to allow ticket holders to enter your event multiple times, but only if the same ticket is not currently checked in to the event.
```

This article tells you how to install pretixSCAN for **Android** and how to make use of its features.

## Prerequisites

pretixSCAN is intended for use with an event hosted with pretix.

You need to have access to a device running Android version 7 or newer.
See our [Android version support policy](https://docs.pretix.eu/en/latest/user/android-version-support.html#pretixscan) for more information.

In order to set up pretixSCAN on your device, you need a QR code or access token generated by the pretix backend.
If you do not have access to the pretix backend yourself, then you need to ask someone who does have that access to send you the QR code or access token.

## General usage

This section guides you through the basic usage of pretixSCAN on Android.
This includes the following steps:

 - [installing the app](#installation-and-setup) and setting it up
 - [connecting to the backend](#connecting-the-device-to-the-pretix-backend)
 - performing basic [entry scanning](#entry-scanning)
 - searching for [attendee data](#searching-for-attendee-data)
 - enabling [badge printing](#enabling-badge-printing)

This section describes those steps in detail.

### Installation and setup

> !!! Note

Reviewer: > !!! Note

Hat das wirklich die Wichtigkeit einer _note_? 
Was kann hier schlimmstenfalls passieren?

Author: Finde die _Note_ hier angemessen, da die durch die visuelle Hervorhebung direkt deutlich macht, dass der Abschnitt für viele Lesende gar nicht relevant ist. 

Das ist auch kein Teil der Anweisungen im Fließtext, daher ergibt die Abspaltung in die _Note_ Sinn.

    pretixSCAN comes preinstalled on the scan smartphones that are available for rent on our website.
    Thus, you will not have to install pretixSCAN on a scan smartphone that you rented from us.
    For more information on our hardware offers, visit [our website](https://pretix.eu/about/en/hardware/scan).

You can install pretixSCAN on your Android device [through the Google Play Store](https://play.google.com/store/apps/details?id=eu.pretix.pretixscan.droid) like any other app.
If your device does not have access to the Google Play Store, you can download the latest version from our [pretix Marketplace](https://marketplace.pretix.eu/products/pretixscan-android/versions).
On devices manufactured by Sunmi, you can also find the app in the Sunmi App Store.

![Screen titled "Welcome to pretixSCAN", asking you to connect the device to the backend.](../../assets/screens/scan/welcome.png "Welcome to pretixSCAN screenshot")

When you launch pretixSCAN for the first time, confirm that you understand the privacy and security implications of storing attendee data on your device.

> You have to grant pretixSCAN access to the device's camera in order to use the core functionality of validating tickets.

Reviewer: > You have to grant pretixSCAN access to the device's camera in order to use the core functionality of validating tickets.

Für Geräte mit ernsthafter scan engine nur bedingt wahr (auch wenn die permission da trotzdem Sinn ergeben kann). 
Das Problem, was ich durch den Hinweis sehe, ist der schon mal angesprochene Fall von "Kunde scannt trotz vorhandener scan engine dennoch über die Kamera". 
Können wir hier den Hinweis einbauen, dass das bei vorhandener Alternative nicht empfehlenswert ist (wenn dir nichts schöneres einfällt)?

Once you have granted permission, pretixSCAN will start accessing the scanner or camera instantly.

> It will also display a text box telling you to create a new device in our organizer account in the pretix backend.

Reviewer: ```suggestion
It will also display a text box telling you to create a new device in your organizer account in the pretix backend.
```

Click the :btn-icon:fa3-plus: Connect a device: button.
Give the device a unique and recognizable name such as "Entrance B phone 1".

If you want to use the device for all your events, check the box next to "All events (including newly created ones)".
If you want to use the device for a limited selection of events, do not check that box.
Instead, select those events under "Limit to events".

Open the "Security profile" drop-down menu and select `pretixSCAN`.

!["Connect to device" page, displaying instructions for installing pretixSCAN, a QR code, and a token.](../../assets/screens/scan/connect-device.png "Connect to device screenshot")

Once you click the :btn:Save: button, pretix redirects you to a page displaying a QR code.
Open pretixSCAN and scan the QR code with the camera or hardware scanner.

If you cannot scan the QR code, then you can connect the device manually instead.
In order to do so, tap the :btn-icon:fa3-ellipsis-v:: three-dot button in the top right corner of the pretixSCAN app.
Then, tap :btn:Manual setup:.
Enter the system URL and token displayed in the pretix backend below the QR code.

> If you want to grant authorization to a device belonging to a person who does not have access to the pretix backend, send them the QR code.

Reviewer: > If you want to grant authorization to a device belonging to a person who does not have access to the pretix backend, send them the QR code.

Ich verstehe den Gedanken, aber ein QR-Code **auf** dem Gerät, das verknüpft werden soll, ist nicht besonders nützlich ... bzw. nur in Kombination mit einem zweiten Endgerät.

Alternatively, send them the system URL and token.

![Screen in pretixSCAN, asking you to choose the event for which you want to scan tickets.](../../assets/screens/scan/select-event.png "Select event screenshot")

> Once you scanned the QR code or entered the token successfully, the app will ask you to select the event for which you want to scan tickets.

Reviewer: ```suggestion
Once you have scanned the QR code or entered the token successfully, the app will ask you to select the event for which you want to scan tickets.
```

> Your next step depends on the type of event for which you want to scan tickets.

Reviewer: > Your next step depends on the type of event for which you want to scan tickets.

In diesem Schritt gibt es eine neue Voraussetzung (die es iirc für das bloße Verknüpfen der Geräte nicht gibt): eine Veranstaltung, die nicht offline ist

 - **Single event:** Select the event in the list or in the calendar and tap the :btn:OK: button.
   Then, select a check-in list and tap the :btn:OK: button again.

>  - **Event series:** Select any of the dates from that event series.

Reviewer: ```suggestion
 - **Event series:** Select one of the dates from that event series.
```

"any of the dates" liest sich für mich wie "ggf. mehrere", was ohne Expertenmodus nicht möglich ist.

   Tap the :btn:OK: button, select the check-in list, and tap the :btn:OK: button again.

 - **Multiple events or series:** Enable expert mode.
   See [Scanning tickets for multiple events](#scanning-tickets-for-multiple-events).

The app will now land you on the main screen and start downloading event data from the server in the background.

### Entry scanning

By default, the app is now in entry scanning mode.
If it is, then it displays a pictogram of a gray box with an arrow pointing to the right :fa3-sign-in: in the top right corner.
If the pictogram is yellow and the arrow points to the left :fa3-sign-out:, that means that the app is in "exit scanning" mode.
Switch to entry scanning by tapping the three-dot button :btn-icon:fa3-ellipsis-v:: and :btn:Switch to entry scanning: in the popup menu.

Our scan smartphones have both a camera and a scanner.
The scanner works much more quickly and reliably than the camera.
Thus, you should always use the scanner at your check-in.

> If you are using a device without a scanner, then you should use the camera instead.

Reviewer: > If you are using a device without a scanner, then you should use the camera instead.

Das ist technisch gesehen korrekt, aber viel Wahlmöglichkeit bleibt da auch nicht, oder? ;)

> In order to confirm that the device uses the right setting, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner, and then tap :btn:Settings:.

Reviewer: ```suggestion
In order to confirm that the device is using the right setting, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner, and then tap :btn:Settings:.
```

Scroll to the section titled "User interface".
If your device has a scanner, uncheck the box next to "Use device camera".
If your device does have a scanner, check that box.

> The scanner is located on the top edge of the device.

Reviewer: > The scanner is located on the top edge of the device.

Fast beliebig an dieser Zeile statt an einer vorigen - hier fiel es nur besonders auf: 
M.-W. scheint dazu zu tendieren, dass "scanner" eher the whole device ist statt nur der von dir gemeinte Teil.

Author: Hast du einen Vorschlag für die Terminologie? Das ganze Gerät nennen wir in diesem Artikel "scan smartphone". Daran würde ich auch festhalten, um den Unterschied zum Handscanner klar zu halten.

Reviewer 2: Zebra nennt den Bereich des Smartphones in seinem englischen Handbuch "Scanner Exit Window". Finde ich eigentlich ganz passend

 2. If the ticket is valid but has already been checked in and not checked out again, pretixSCAN displays a yellow box titled "Ticket already used".
 Underneath the yellow box, it lists the ticket type, order number, and position number, and also the date and time when the ticket was first scanned.
 3. In all other cases, pretixSCAN displays a red box with a title explaining the problem.
 Explanations can include, among others, "Invalid ticket", "Order canceled", or "Entry not allowed".

![Scanning a ticket in pretixSCAN. A green box appears at the top of the screen, displaying a checkmark, the text: "Valid ticket (Speaker ticket)" and the order code suffixed with "-1".](../../assets/screens/scan/valid.png "Valid ticket screenshot")

If none of these things happen, then your device has not scanned the code.
Try adjusting the angle and distance between scanner and ticket or improving ambient lighting.
You can turn the device's flashlight and automatic focus on or off with the buttons at the bottom of the screen.
If you have trouble scanning a code from a phone screen, ask the customer to increase screen brightness to maximum.

> You can also try tilting the scanning device by 90 degrees in any direction.

Reviewer: > You can also try tilting the scanning device by 90 degrees in any direction.

... im Falle spontan auftretender temporärer Erblindung sollten Sie eine andere Richtung versuchen.

### Searching for attendee data

An alternative to scanning a ticket code is searching for an attendee in pretixSCAN.
If an attendee arrives at the check-in without a ticket, but claiming that they have bought one, you can search for their data.
Tap the :btn-icon:fa3-search:: button at the top of the screen.
In the search field, enter the attendee's name, order code, ticket secret, or email address.
Tap the matching result in the list.

The app will now check the validity of the ticket and return the same results as if it was scanning the ticket.

### Enabling badge printing

Badge printing only works if you have enabled the "Badges" plugin for the current event and the ticket you are scanning has a badge layout selected.
In order to enable badge printing in pretixSCAN, you have to first install our additional app pretixPRINT on the same device and configure a connection to a printer.

> !!! Note

Reviewer: > !!! Note

Warum kommt dieser Hinweis erst hier bzgl. pretixPRINT und nicht bereits früher bzgl. Einrichtung von SCAN/Verknüpfung mit dem Backend?
Für letzteres ist das mit noch höherer Wahrscheinlichkeit wahr als für pretixPRINT:
Es kommt vor, dass im Nachhinein noch weitere Scan-Smartphones hinzugemietet werden, für die wir die Einrichtung von pretixPRINT dann nicht vornehmen können, da der Etikettendrucker bereits vor Ort ist.

    pretixPRINT comes preinstalled and configured on the scan smartphones that are available for rent on our website.
    Thus, you will normally not have to install and configure pretixPRINT on a scan smartphone that you rented from us.

Once you have done that, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner, and then tap :btn:Settings:.

> Scroll down to the section titled "Badges" and check the box next to "Enable badge printing".

Reviewer: > Scroll down to the section titled "Badges" and check the box next to "Enable badge printing".

Ich würde "Print badges automatically" etc. als Untermenüpunkte hiervon betrachten; du nicht?

Author: Grenzfall. Die Überschrift ist "Badges", aber die beiden unteren Menüpunkte sind ausgegraut, solange "Enable badge printing" nicht aktiviert ist. 
Ich sehe gerade nicht, was das am Text ändern würde

Reviewer: Habe ich mir das eingebildet? Kam das Thema "Print badges automatically" hier nicht vor?

The app will still display the :btn-icon:fa3-print:: button, allowing you to manually print an extra badge.

If you want to print out all badges twice after a scan, open the settings in pretixSCAN.
Scroll down to the section titled "Badges" and check the box next to "Print every badge twice".

### Switching to a different event or check-in list

If you want to switch to a different event or to a different check-in list for entry scanning, tap the name of your event at the top of the screen.
Select the event from the calendar or list and tap the :btn:OK: button.
Select the check-in list and tap the :btn:OK: button again.

If you want to scan tickets for multiple events at the same time, you should enable **expert mode**.
The same applies if you want to scan tickets for multiple dates in an event series which all have their own check-in lists.
See [Scanning tickets for multiple events](#scanning-tickets-for-multiple-events) for more information.

## Advanced usage

This section covers the following advanced use cases of pretixSCAN on Android:

 - [exit scanning](#exit-scanning)
 - [locking the settings](#locking-the-settings)

>  - Letting attendees [scan their own tickets](#letting-attendees-scan-their-own-tickets) using kiosk mode

Reviewer: ```suggestion
 - letting attendees [scan their own tickets](#letting-attendees-scan-their-own-tickets) using kiosk mode
```

Alle anderen Stichpunkte beginnen gerade auch mit Kleinbuchstaben

 - [offline scanning](#offline-scanning)
 - scanning tickets for [multiple events](#scanning-tickets-for-multiple-events)

### Exit scanning

If you want to record ticket holders leaving your event, you can use pretixSCAN's exit mode.
This is useful if you want to allow ticket holders to enter your event multiple times, but only if the same ticket is not currently checked in to the event.

> It is also useful if you want to add tickets that have attended the event and then left back to the quota, thus allowing you to sell more tickets.

Reviewer: > It is also useful if you want to add tickets that have attended the event and then left back to the quota, thus allowing you to sell more tickets.

Die Tickets selbst besuchen und verlassen eher nicht die Veranstaltung - bekommen wir das mit vertretbarem Aufwand umformuliert?

Search for the device for which you want to enable the kiosk mode security profile in the list.
Click the :btn-icon:fa3-edit:: "edit" button next to it.
On the next page, under "Security profile", select `pretixSCAN (kiosk mode, no order sync, no search)`.
Click the :btn:Save: button.

!!! Note
    The kiosk mode security profile disables order search.
    If you want the possibility to use the order search on this device, select the security profile `pretixSCAN` instead.
    If you follow the rest of the instructions in this section, then order search will still be locked behind a PIN.

In order to enable kiosk mode on your device, lock the settings behind a PIN as described under [Locking the settings](#locking-the-settings).
While you are in the "PIN protection" submenu, check the box next to "Kiosk mode".
Once you return to the main screen, pretixSCAN will not display the menu bar at the top anymore.

![Home screen of pretixSCAN in kiosk mode displaying the pictogram for entry scanning, the last synchronization, and the buttons for flashlight and autofocus. The top bar is missing.](../../assets/screens/scan/kiosk.png "Kiosk mode")

> Use the [App pinning](https://support.google.com/android/answer/9455138?hl=en) feature of Android or your preferred solution to prevent users from leaving the app.

Reviewer: ```suggestion
Use the [app pinning](https://support.google.com/android/answer/9455138?hl=en) feature of Android or your preferred solution to prevent users from leaving the app.
```

It will still occasionally attempt to synchronize its own internal data with the server.
You can scan codes and search for attendee data without encountering errors despite the lack of a network connection.

Installing the app, connecting it to the backend and synchronizing data between the device and the server at least once still requires a working connection to the internet.
Do these steps ahead of time in a setting where you do have a reliable connection if you are planning to use pretixSCAN in offline mode.

!!! Warning
    If you are using offline scanning with more than one device, then attendees can potentially use a single ticket for entry more than once.
    However, this is only possible for a limited period of time.
    As soon as the device that has scanned the ticket performs a successful synchronization, and then the other devices also perform a successful synchronization.
    After that, pretixSCAN will correctly identify the ticket as "Ticket already used".

If you want to enable offline mode, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner and then tap :btn:Settings:.

> Then check the boxes next to "Offline scanning", "Automatic Synchronization", and "Download orders".

Reviewer: > Then check the boxes next to "Offline scanning", "Automatic Synchronization", and "Download orders".

Das klingt nach 
"1. Offline scanning und 
2. Automatic Synchronization (very online) 
anschalten". 
Können wir den Widerspruch auflösen?

If the network connection at your check-in is present, but unstable, you can leave "Offline scanning" unchecked and instead tap :btn:Automatically turn offline mode on and off:.
The default selection is `Manual mode (off)`.
This means that pretixSCAN will not enable or disable offline scanning automatically.

![Settings page in pretixSCAN. 'Offline scanning' is disabled. 'Automatically turn offline mode on and off' is set to '3 seconds or errors'.](../../assets/screens/scan/offline.png "Automatically turn offline mode on and off")

If you select, for instance, "3 seconds or errors", pretixSCAN will enable offline scanning if at least one of the following three conditions is met:

 - pretixSCAN has been attempting to verify a code for three seconds without success
 - pretixSCAN has encountered three errors while attempting to verify a code
 - pretixSCAN has detected a loss of connection while attempting to verify a code

If you select "Only errors or connection loss", then pretixSCAN will enable offline scanning after encountering an error or detecting a loss of connection while attempting to verify a code.

!!! Note

>     In offline mode, pretixSCAN will not recognize any tickets ordered while your shop is in test mode as valid.

Reviewer: >    In offline mode, pretixSCAN will not recognize any tickets ordered while your shop is in test mode as valid.

_Sollte_ das im Realbetrieb passieren, kennen wir da einen Anwendungsfall für?

Author: Das kann halt beim Testen des ganzen Aufbaus (App, Einlassregeln, sonstige Konfiguration in pretix, Internetverbindung etc.) auffallen. Fand es daher wichtig, es hier zu erwähnen. Mich selbst hat es auch kurz verwirrt.

Reviewer 2: Das passiert auch Kund*innen und es gibt deswegen Anrufe/Tickets.

    Also, tickets ordered in test mode will not appear in the search.

### Scanning tickets for multiple events

By default, pretixSCAN scans tickets for one event at a time.
If you want to scan tickets for more than one event, you can use **expert mode**.

This is useful if you are operating a single check-in for multiple events that are taking place at the same time.

If you want to scan tickets for multiple events or event series, then you need to give your device permission for all of those.
In order to do so, open the [pretix backend](https://pretix.eu/control/) and navigate to :navpath:Your organizer → :fa3-mobile-phone: Devices:.

Search for the device to which you want to give permission.

> Click the :btn-icon:fa3-edit:: "edit" button next to it.

Reviewer: > Click the :btn-icon:fa3-edit:: "edit" button next to it.

Ich verstehe die Logik, aber ich finde es nicht völlig sinnvoll. Ein Gerät lässt sich ja genausogut bei der initialen Verknüpfung mit mehreren Veranstaltungen verbinden. 
Der Aufbau hier erweckt den Eindruck, als sei es zwingend notwendig, zuerst mit einer einzelnen Veranstaltung zu verknüpfen und dann zu editieren. Dies ist nicht der Fall.

On the next page, if you want to give the device permission for all events, check the box next to "All events (including newly created ones)".
If you want to give permission for a limited selection of events, under "Limit to events", select the events for which you want to scan tickets.
Click the :btn:Save: button.

> In order to activate expert mode in pretixSCAN, tap the name of your event in the bar at the top.

Reviewer: > In order to activate expert mode in pretixSCAN, tap the name of your event in the bar at the top.

Dieser Schritt ist in der Form nur wahr, wenn bereits eine Veranstaltung ausgewählt wurde.

Author: Wenn man die App gerade erst verbindet, wird man durch das Scannen des QR-Codes oder durch die manuelle Einrichtung direkt in die Auswahl der Veranstaltung geworfen. Also erübrigt sich die Alternative hier

Select one of the events for which you want to scan tickets.
Then, tap the :btn:OK: button.
Select your check-in list.
Check the box next to "Expert mode (select multiple events)" and tap the :btn:OK: button.

![Page titled 'Select Check-in list' in pretixSCAN. The box next to 'Expert mode (select multiple events)' is checked.](../../assets/screens/scan/expert-mode.png "Expert mode")

Tap the :btn-icon:fa3-plus:: button.
Select another event and check-in list for which you want to scan tickets.
Repeat this step for every event for which you want to scan tickets.
You can add as many events as you like.

> But you can only add one check-in list per event

Reviewer: ```suggestion
But you can only add one check-in list per event.
```

Würde auch in Erwägung ziehen, das mit dem vorigen Satz ("You can add as many events as you like") zusammenzulegen

We recommend using the scanner.

Once you have granted permission, pretixSCAN will start accessing the scanner or camera instantly.
It will also display a text box telling you to create a new device in your organizer account in the pretix backend.
The next section is going to tell you how to do that.

### Connecting the device to the pretix backend

!["Connect a new device" page, showing options for the name of the device, the events it should have access to, its security profile, and gate.](../../assets/screens/scan/connect-new-device.png "Connect new device screenshot")

Open the [pretix backend](https://pretix.eu/control/) and navigate to :navpath:Your organizer → :fa3-mobile-phone: Devices:.
Click the :btn-icon:fa3-plus: Connect a device: button.
Give the device a unique and recognizable name such as "Entrance B phone 1".

If you want to use the device for all your events, check the box next to "All events (including newly created ones)".
If you want to use the device for a limited selection of events, do not check that box.
Instead, select those events under "Limit to events".

> Open the "Security profile" drop-down menu and select `pretixSCAN`.

Reviewer 2: Should we maybe list all security profiles, their use cases and implications here?
For example one might like to use the 'pretixSCAN online only' profile to make sure that no user data is stored on the device (at least not in bulk beforehand like if you synchronize) and the person using pretixSCAN can't search the data.

Author: Search still works. It doesn't work if you select `pretixSCAN (kiosk mode, no order sync, no search)`, as advertised. 
I noticed a different issue with this setting though. Opening that can of worms in our slack.

You can print the code and hand it to them, or send it to a separate device.
It makes no sense to send the QR code to the device running pretixSCAN.
As an alternative to the QR code, you can send them the system URL and token.

![Screen in pretixSCAN, asking you to choose the event for which you want to scan tickets.](../../assets/screens/scan/select-event.png "Select event screenshot")

Once you have scanned the QR code or entered the token successfully, the app will ask you to select the event for which you want to scan tickets.
It is not possible to select an event that is currently offline.
Your next step depends on the type of event for which you want to scan tickets.

 - **Single event:** Select the event in the list or in the calendar and tap the :btn:OK: button.
   Then, select a check-in list and tap the :btn:OK: button again.

 - **Event series:** Select one of the dates from that event series.
   Tap the :btn:OK: button, select the check-in list, and tap the :btn:OK: button again.

 - **Multiple events or series:** Enable expert mode.
   See [Scanning tickets for multiple events](#scanning-tickets-for-multiple-events).

> The app will now land you on the main screen and start downloading event data from the server in the background.

Reviewer 2: Not true if the security profile is not pretixSCAN. Should we maybe add the prerequisite?

### Entry scanning

By default, the app is now in entry scanning mode.
If it is, then it displays a pictogram of a gray box with an arrow pointing to the right :fa3-sign-in: in the top right corner.
If the pictogram is yellow and the arrow points to the left :fa3-sign-out:, that means that the app is in "exit scanning" mode.
Switch to entry scanning by tapping the three-dot button :btn-icon:fa3-ellipsis-v:: and :btn:Switch to entry scanning: in the popup menu.

Our scan smartphones have both a camera and a scanner.
The scanner works much more quickly and reliably than the camera.
Thus, if you have a scanner, then you should always use the scanner at your check-in.

In order to confirm that the device is using the right setting, open pretixSCAN, tap the three-dot button :btn-icon:fa3-ellipsis-v:: in the top right corner, and then tap :btn:Settings:.
Scroll to the section titled "User interface".
If your device has a scanner, uncheck the box next to "Use device camera".
If your device does have a scanner, check that box.

The scanner is located on the top edge of the device.
The buttons for activating the scanner are on the left and right edges.

> In order to scan a QR code or barcode with the scanner, point the scanner at the code and press one of the buttons.

Reviewer 2: pretix doesn't generate barcodes, thus I would remove that term. And after seeing many of our customers using our scan smartphones at a much too short distance like a very near sighted little hamster I would add sth. like

```suggestion
In order to scan a QR code with the scanner, point the scanner at the code from a distance of roughly 30-50 cm and press one of the buttons.
```

The camera is located on the back of the device.

> In order to scan a QR code or barcode with the scanner, point the camera at the code so that the code is visible on the screen in pretixSCAN.

Reviewer 2: See above: There are no barcodes with pretix.
```suggestion
In order to scan a QR code with the scanner, point the camera at the code so that the code is visible on the screen in pretixSCAN.
```

 Underneath the yellow box, it lists the ticket type, order number, and position number, and also the date and time when the ticket was first scanned.
 3. In all other cases, pretixSCAN displays a red box with a title explaining the problem.
 Explanations can include, among others, "Invalid ticket", "Order canceled", or "Entry not allowed".

![Scanning a ticket in pretixSCAN. A green box appears at the top of the screen, displaying a checkmark, the text: "Valid ticket (Speaker ticket)" and the order code suffixed with "-1".](../../assets/screens/scan/valid.png "Valid ticket screenshot")

If none of these things happen, then your device has not scanned the code.
Try adjusting the angle and distance between scanner and ticket or improving ambient lighting.
You can turn the device's flashlight and automatic focus on or off with the buttons at the bottom of the screen.
If you have trouble scanning a code from a phone screen, ask the customer to increase screen brightness to maximum.
You can also try tilting the scanning device by 90 degrees in any direction.

### Searching for attendee data

> An alternative to scanning a ticket code is searching for an attendee in pretixSCAN.

Reviewer 2: This won't work if the security profile pretixSCAN online only was selected for the device

Tap the :btn-icon:fa3-plus:: button.
Select another event and check-in list for which you want to scan tickets.
Repeat this step for every event for which you want to scan tickets.
You can add as many events as you like, but you can only add one check-in list per event.

You can change the check-in list by tapping the :btn-icon:fa3-pencil:: button next to it.
You can remove the event by tapping the :btn-icon:fa3-trash:: button.

Once you are happy with your choices, tap the :btn-icon:fa3-check:: button at the top of the screen.
This takes you back to the home screen of pretixSCAN.

If you want to **disable** expert mode, tap the name of your event at the top of the screen.
Remove all events for which you do not want to scan tickets.
Then, tap the three-dot menu :btn-icon:fa3-ellipsis-v:: and tap :btn:Disable expert mode:.

![Page titled 'Select Events' in pretixSCAN. The only entry in the list is 'Tutorial Conference'. There is a button labeled 'Disable expert mode' in the top right corner of the screen.](../../assets/screens/scan/expert-mode-disable.png "Disable expert mode")

## Troubleshooting

> 

Reviewer 2: Additional topic:
### External barcode scanners
External barcode scanners connected via USB act like keyboards: they scan a QR code and send the decoded data as keystrokes. Scanners use keyboard layouts, so if a scanner’s layout differs from the host device’s layout, some characters from the QR code may be decoded incorrectly (for example, a 'z' on a US layout is interpreted as 'y' on a German layout), which can produce invalid ticket codes."

Reviewer 2: Sorry, I got a little bit carried away: Though there are some customers who connect a barcode scanner to a Android POS and use pretixSCAN with this POS device to check in tickets, this mostly is a problem when using pretixSCAN/Linux/Windows

If you want to use the device for a limited selection of events, do not check that box.
Instead, select those events under "Limit to events".

Open the "Security profile" drop-down menu and select one of the options starting with `pretixSCAN`.
There are three such options:

 - `pretixSCAN`: Use this option if you want to use all functions of pretixSCAN.

>  - `pretixSCAN (offline only, no order sync)`: Use this option if you do not want to store order data on the device.

Reviewer 2: >  - `pretixSCAN (offline only, no order sync)`: Use this option if you do not want to store order data on the device.

(online only, no order sync)
